Karttakuvan klikkaus

tatut opened the pull request.

tatut committed:
Karttatason asioiden haku pisteessä

Tason protokollaan uusi metodi, jolla voidaan hakea koordinaatille
löytyneet asiat.

tatut committed:
Asioiden tiedot esimerkki

tatut committed:
Lisää require jotta koodi kääntyy

tatut committed:
Laukaise asioiden haku kartan klikkaamisesta

tatut committed:
Hae asiat pisteessä, kun klikataan

tatut committed:
Lisää haku asioille karttakuvien reksiteröintiin

tatut committed:
Korjaa työkoneenseurannan puhdistus init

erno committed:
Lisää infopaneeli fronttiin (ei vielä sisältöä)

erno committed:
Infopaneelin voi sulkea

tatut committed:
Käytä sijainti, älä pikselikoordinaatteja

erno committed:
Spinner lisätty

tatut committed:
Parametrit mäppinä palvelimelle

tatut committed:
Alustava toteumien asioiden haku

tatut committed:
Palvelu asioiden hakemiseen karttakuvalle

tatut committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Lisää :tyyppi-kartalla avain palautettuihin

erno committed:
Infopaneeli: valittujen asioiden näyttölogiikka

erno committed:
infopaneeli: siistimistä

Kauko committed:
Poista turhia tee-kentta riippuvuuksia

Kauko committed:
Erota kartan tietoja eri namespaceen riippuvuusketjun vuoksi

Uusi infopaneeli käyttää kentät namespacea, jossa oli tierekisteri-
komponentin vuoksi riippuvuus kartat namespaceen, joka taas käyttää
infopaneelia. Uusi tiedot/kartta namespace sisältää kartan
tilaan liittyviä atomeja ja funktioita.

Kauko committed:
Yksityiskohtien esittäminen näyttää kentät skeeman mukaan

Kauko committed:
Näytä koordinaatit paneelissa myös

Kauko committed:
Lisää alustavat overlay skeemat kaikille asioille

erno committed:
Komponenttitesti kartan infopaneelille

erno committed:
syntaksikorjauksia

Kauko committed:
Overlayn tiedot voidaan muodostaa .cljs tiedostossa

Kauko committed:
Varoita jos overlayn muodostamiseen tarvittavia tietoja puuttuu

Kauko committed:
WIP Aseta infopaneeli näkyville karttaa klikattaessa

Kauko committed:
Infopaneeli kuuntelee näkyvyys-atomia

Kauko committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

Kauko committed:
Poista infopaneelin skeemasta tyhjät riviskeemat

Kauko committed:
Lisää alustava infopaneelilinkkien käsittely

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

Kauko committed:
Toivottavasti hieman kuvaavammat nimet

Kauko committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Palauta tarkemmat tehtävän tiedot

Kauko committed:
Pikafiksi infopaneelin näyttämisen hallintaan

Kauko committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Tarkastusten asioiden haku

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

Kauko committed:
Yhdistä infopaneelin aukaiseminen vanhaan on-click/select käsittelyyn

Kauko committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Yhdista karttakuvan ja asioiden haun params

tatut committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Tarkastuksen tiedot kartalle

tatut committed:
Infopaneelin tausta läpinäkyväksi

tatut committed:
Tyyppi keywordina

tatut committed:
Tyylitä infopaneelia

Tehdään infopaneelin tyylit luokan perusteella.
Lisätty otsikolle bold ja isompaa fonttia.

Kauko commented on dev-resources/less/kartta.less:
```diff
@@ -271,7 +271,7 @@
  }
 
-#kartan-infopaneeli {
+.kartan-infopaneeli {
```
Miksi luokka?

tatut replied:
Hyvä kysymys... muistelen Jarnon opeista, että luokkaa olisi parempi käyttää. Mutta ei sillä tässä tapauksessa taida olla väliä, infopaneeleja ei koskaan ole useita näkyvissä.

Kauko commented on dev-resources/less/kartta.less:
```diff
@@ -280,5 +280,5 @@
  min-width: 200px;
  max-width: 1000px;
- background-color: @harmaa6;
+ background-color: @harmaa6tp;
```
Nopealla googletuksella voisi toimia fade(@harmaa6, 66%);

Olisikohan parempi? Jos katsoo livi_tyylejä nyt, niin tommonen @harmaa6tp tuntuu aika maagiselta.

tatut committed:
Paranna kentän otsikon tyyliä

tatut committed:
Käytä remove ikonia "x" kirjaimen sijasta

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Käytä fade transparent värin sijasta

tatut committed:
Infopaneeli oletuksena kiinni

tatut committed:
Korjaa mergehärö

tatut committed:
Dummy haku tilannekuvan tarkastuksille

erno committed:
Debug-viesti puutteelliselle skeemalle

erno committed:
infopaneeli: tyylejä ja ulkoasua

tatut committed:
Tilannekuvan tarkastusten haku pistellee

tatut committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

erno committed:
infopaneelin näyttölogiikka

erno committed:
nimiviilaus

erno committed:
merge-korjaus

tatut committed:
Muuta nimet spesifisiksi

Koodikahveilla käyty läpi nimeämistä ja todettu spesifimmät nimet paremmiksi

tatut committed:
Muuta tiedoston nimi vastaamaan ns nimeä

tatut committed:
Nimiparannuksia koodikahvien perusteella

tatut committed:
Sulje infopaneeli, kun karttatasot muuttuvat

tatut committed:
Korjaa require

tatut committed:
Poista kuollutta koodia

tatut committed:
Korjaa assert

tatut committed:
Näytä klikatulle pisteelle neula

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

Kauko committed:
Paranna infopaneelin error-logitusta

Aiemmin skeemat, joissa ei ollut :nimi tai :hae kenttiä,
logitettiin tyhjinä arvoina varoituksessa, joka ilmoittaa datasta
puuttuvista kentistä.

erno committed:
infopaneeli: tasoita toteuma-skeema

Kauko committed:
Openlayers ei tarvi enää :asiat-pisteessä atomia

Kauko committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

erno committed:
flatten -> concat

Kauko commented on src/cljs/harja/ui/kartta/infopaneelin_sisalto.cljs:
```diff
@@ -163,30 +163,30 @@
-                          (get-in % [:materiaalit materiaalitoteuma :materiaali :yksikko]))})
-            (when (:lisatieto toteuma)
-              {:otsikko "Lisätieto" :nimi :lisatieto})]
+   :tiedot (vec (flatten [{:otsikko "Alkanut" :tyyppi :pvm-aika :nimi :alkanut}
```
Jos tasoituksen tekee tällä tasolla, niin käyttäisin mieluummin sitä ehdottamaasi concattia. Flattenin etu olis se, että sen voisi laittaa tuonne validoi-tieto funktioon.

Koska toteumat on ainoa joka pitää tällä hetkellä tasoittaa, niin ehdottamasti concat on parempi mielestäni.

erno committed:
infopaneeli: default-metodi infopaneeli-skeema -multimetodille

erno committed:
infopaneeli: poista asioiden tuplaus (oli testiksi)

erno committed:
infopaneeli: lisää virhelogitus

erno committed:
infopaneeli: palauta nil virhetapauksessa

tatut committed:
Työkoneseurantataulu rivi per havainto

Muutetaan työkoneseurantataulua palvelemaan uutta seurantaa, jossa
jokainen havainto halutaan talteen, ei vain viimeisin sijainti.

Lisäksi indeksoidaan havainnot sijainnin perusteella nopeampaa hakua varten.

erno committed:
infopaneeli: vertaile @valittu-asia ilman mapin funktiokenttiä

erno committed:
infopaneeli: poista tilannekuvan testilinkki

erno committed:
kytke infopanelein komponenttitesti päälle runnerissa, ja laajenna
hieman testiä

tatut committed:
Työkoneseurannan uudet kyselyt

tatut committed:
Käytä mäp muotoa työkonehavainnon tallennuksessa

tatut committed:
Korjaa viivojen piirron dash pattern

Viivoja ei enää piirretä erillisinä segmentteinä vaan yhtenä pathina,
jolloin dash pattern toimii eikä ala alusta joka segmentille.

tatut committed:
Pura primary key constraint

tatut committed:
Tee työkoneista palvelinpään karttataso

tatut committed:
Lisää nimeen <!, koska insert

tatut committed:
Työkoneen karttakuva

tatut committed:
Työkoneen selite

tatut committed:
Serverille parempi työkoneen dash pattern

tatut committed:
Poista kommentoitu logitus

tatut committed:
Käytä viivan taitoksia työkoneen viivalle

tatut committed:
Merge branch 'tyokoneiden_havainnot_aikaleimoilla' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Korjaa mergehärö

tatut committed:
Poista käyttämätön palvelu ja sen testi

tatut committed:
Älä käytä poistettua komponenttia

erno committed:
siistimistä tyyliin

erno committed:
kierrä phantomjs-bugi console.logissa

erno committed:
korjaa infopaneelin testi ja päivitä testidata

erno committed:
infopaneeli: työkoneiden tietojen haku backendissä

erno committed:
funktio tilannekuvan suodatinten parsintaan

erno committed:
infopaneeli/tilannekuva/työkoneet: käsittele tehtavat-array
paremmin, korjaus suodatinparsintaan

erno committed:
infopaneeli: työkoneiden työ alkanut-tieto

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Korjaa työkoneseurannan testi

tatut committed:
Hae työkoneet vain nykytilanteeseen

tatut committed:
Käytä uutta palautusmuotoa

Työkoneet renderöidään palvelimella, joten frontille palautetaan vain
tehtäväsetti alueelta löytyneistä työkoneista.

tatut committed:
Korjaa näkyvyystesti

Testataan että selitteissä on oikeat asiat (ei spesifistä työkonetta,
koska ne piirretään serverillä).

tatut committed:
Kok.hint. toteuman tiedot infopaneeliin

Toteuta kok.hint. toteuman tietojen haku infopaneeliin.

tatut committed:
Yhtenäistä yht/kht toteuman tietokyselyä

tatut committed:
Hae yht oikein infopaneeliin

tatut committed:
Muuta migraationumero

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Näytä toteuman määrät oikein

tatut committed:
Yhtenäistä transit parametreja

Kaikki tasot käyttivät transit enkoodattua erikoisparametria, josta
haettiin arvoja kyselyyn. Nyt karttakuvan rekisteröinnissä voi antaa
parametrin nimen ja ne arvot yhdistetään muihin parametreihin. Näin
transit lukemista ei tarvitse kaikkien karttakuvatasojen itse toteuttaa.

tatut committed:
infopaneeli-skeema eri ilmoitustyypeille

tatut committed:
Paranna paneelin/neulan piilotuslogiikkaa

Enää ei tarvita go-lohkoa hoitamaan asynkronisesti. Karttageometrioiden
muutoskuuntelija tunnistaa, jos jotain muuta kuin näkymän geometriat
ovat muuttuneet, ja piilottaa vain siinä tapauksessa.

erno committed:
infopaneeli: tarkastusten linkkinapit

tatut committed:
logita warn puuttuvasta toteutuksesta

Jos infopaneelissa yritetään näyttää jotain, mille ei ole määritelty
toteutusta infopaneeli-skeema multimetodille, tehdään warn logitus.

erno committed:
infopaneelii: korjaus linkkiklikkaukseen

tatut committed:
Käytä wrap atomin sijasta

Käytetään wrapia, joka logittaa virheen, jos infopaneelitietoa yritetään kirjoittaa.

tatut committed:
Lisää silloille tyyppi-kartalla

tatut committed:
Lisää sillalle infopaneeli-skeema

tatut committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

erno committed:
viittaa navigaatio-nimiavaruuteen yhdellä nimellä

tatut committed:
Avaa silta infopaneelista

tatut committed:
Typo fix turvallisuuspoikkeama

Lisää puuttuva 'u'

tatut committed:
Vammoja vain 1

erno committed:
kokonaishintaiset toteumat WIP

erno committed:
uudelleennimeä kaima-ns viittaus -> tiedot

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

erno committed:
klikattavat ykiskkökohtaiset tiet WIP

erno committed:
Merge remote-tracking branch 'origin/karttakuvan_klikkaus' into kk2

tatut committed:
Korjaa parametrit infopaneelihaussa

tatut committed:
Merge branch 'karttakuvan_klikkaus' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Poista kuollutta koodia

tatut committed:
Poista parametrien debug printtaus

tatut committed:
Näytä toteumien ja tarkastusten TR-osoite

Kun klikkauspisteen toteuma/tarkastus haetaan, yritetään hakea TR-osoite
reittigeometrian ensimmäisen/viimeisen pisteen mukaan.

tatut committed:
Poista toteuma_id summarivistä

Summarivit ovat saman päivän kaikkien toteumien yhteenlaskettuja määriä
tietylle toimenpiteelle.

tatut committed:
Yksittäisen toteuman valinta infopaneelista

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Poista turha :linkki kenttä

tatut committed:
Turpon avaus infopaneelista

erno committed:
infopaneeli: näytä saman työkoneen työkonehavainnot yhtenä rivinä

tatut committed:
Yksinkertaista geometrian klikkauslogiikkaa

tatut committed:
Tee errorista normaali logitus

tatut committed:
Korjaa testin magic numberit

tatut committed:
Laske klikkaustoleranssi extentistä

Lasketaan klikkaustoleranssi extentin hypotenuusan pituuden suhteessa
eikä käytetä 150m kovakoodattua arvoa.

tatut committed:
Käytä valmiiksi laskettua toleranssia

tatut committed:
Käytä laskettua toleranssi parametria

tatut committed:
Avaa yks.hint. toteuma linkistä

tatut committed:
Muuta error normaaliksi logitukseksi

tatut committed:
Lisää marginia infopaneelin linkkinapille

tatut committed:
Tee otsikosta klikattava

tatut committed:
Ilmoituksen näyttö infopaneelissa tilannekuvassa

tatut committed:
Paranna työkoneen ja toteuman otsikoita

tatut committed:
Muuta migraationumero

tatut committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into karttakuvan_klikkaus

tatut committed:
Korjaa mergehärö
